### PR TITLE
feat(profiling): add transaction table sorting

### PR DESCRIPTION
**Pull request by eliashussary**
This PR introduces sorting capability for the Transaction table present on the profiling page;

image
image
image

Closes: https://github.com/getsentry/team-profiling/issues/56

### CODE REVIEW

File: static/app/utils/profiling/tableRenderer.tsx [hunk 1]
```diff
@@ -20,17 +16,18 @@ export function renderTableHead<K>({
...
+        onClick={e => onClick?.(column, e)}
         align={rightAlignedColumns?.has(column.key) ? 'right' : 'left'}
         title={column.name}
-        direction={
-          currentSort?.column === column.key ? currentSort?.direction : undefined
-        }
+        direction={currentSort?.key === column.key ? currentSort?.order : undefined}
         canSort={sortableColumns?.has(column.key) || false}
         generateSortLink={generateSortLink?.(column.key) ?? (() => undefined)}
+        replace
```
Comment (JonasBa) [inline static/app/utils/profiling/tableRenderer.tsx:30]: What do you think about handling this from the onClick vs introducing a new prop?

**Review by JonasBa — APPROVED** [review]
Lgtm, just left a small comment but not blocking

File: static/app/utils/profiling/tableRenderer.tsx (hunk 1 above)
Comment (eliashussary) [reply static/app/utils/profiling/tableRenderer.tsx:30]: i had thought about it and figured it'd be better to align to `react-router`'s more to date api; https://v5.reactrouter.com/web/api/Link/replace-bool  ... we're obviously a couple versions behind and its not currently supported

my other thought around it was that this `SortLink` takes a `generateSortLink`,  with the intent of it being responsible for the actual routing action. by moving the routing to an imperative call inside `onClick` it overrides the utility of `SortLink` and it just becomes a pure UI component.  

i'm good with either way

i found parts of Grid api weird; this is a good example

Comment (JonasBa) [reply static/app/utils/profiling/tableRenderer.tsx:30]: Good idea with the forward compatibility 🙏🏼 (I like the imperative call too, but this is not a big deal either way)

**Review by JonasBa — APPROVED** [review]

**Pull request merged**